pkg/sources/adapter/webhooksource/webhook.go

```diff
@@ -194,10 +194,15 @@ func (h *webhookHandler) handleAll(ctx context.Context) http.HandlerFunc {
 			return
 		}
 
-		if result := h.ceClient.Send(ctx, event); !cloudevents.IsACK(result) {
+		rEvent, result := h.ceClient.Request(ctx, event)
+		if !cloudevents.IsACK(result) {
 			h.handleError(fmt.Errorf("could not send Cloud Event: %w", result), http.StatusInternalServerError, w)
+			return
+		}
+		if rEvent.Data() == nil {
```

Comment:
Avoid panics when no response is returned

```suggestion
		if rEvent == nil || rEvent.Data() == nil {
```